src/parser/jobs/mnk/modules/RiddleOfFire.tsx

```diff
@@ -1,198 +1,262 @@
+import {GlobalCooldown} from 'parser/core/modules/GlobalCooldown'
+import {SEVERITY, Suggestion, TieredSuggestion} from 'parser/core/modules/Suggestions'
 import React from 'react'
 import {BLITZ_SKILLS} from './constants'
 import {DISPLAY_ORDER} from './DISPLAY_ORDER'
 import {fillActions} from './utilities'
 
-// Expected GCDs and buffs
-// technically can get 2 tacles and should as much as possible, but don't ding for it in case it's for mechanics
-const EXPECTED = {
-	GCDS: 11,
-	BLITZES: 1,
+const SEVERITIES = {
+	TOTAL_GCD_DURING_ROF_SEVERITY: {
+		10: SEVERITY.MINOR,
+		9: SEVERITY.MEDIUM,
+		6: SEVERITY.MAJOR,
+	},
+	BROTHERHOOD_OUTSIDE_ROF_SEVERITY: {
+		1: SEVERITY.MEDIUM,
+		2: SEVERITY.MAJOR,
+	},
+	BAD_GCD_DURING_ROF_SEVERITY: {
+		1: SEVERITY.MEDIUM,
+		2: SEVERITY.MAJOR,
+	},
 }
 
-const SUGGESTION_TIERS = {
-	1: SEVERITY.MINOR,
-	2: SEVERITY.MEDIUM,
-	5: SEVERITY.MAJOR,
-}
+const DOUBLE_PERFECT_BALANCE_IN_SECONDS = 80
+
+const EXPECTED_GCDS = 11
+
+class MasterfulBlitzEvaluator implements WindowEvaluator {
+	private blitzSkills: Array<Action['id']>
+	private celestialRevolution: Action
+	private brotherhood: Action
 
-class Riddle {
-	data: Data
-	casts: Array<Events['action']>
-	start: number
-	end?: number
-	rushing: boolean = false
-
-	constructor(start: number, data: Data) {
-		this.data = data
-		this.start = start
-		this.casts = []
+	constructor(blitzSkills: Array<Action['id']>, celestialRevolution: Action, brotherhood: Action) {
+		this.blitzSkills = blitzSkills
+		this.celestialRevolution = celestialRevolution
+		this.brotherhood = brotherhood
 	}
 
-	get gcds() {
-		return this.casts
-			.filter(event => this.data.getAction(event.action)?.onGcd ?? false)
+	suggest(windows: Array<HistoryEntry<EvaluatedAction[]>>) {
+		const celestialRevolutionUsages = windows.map(previousValue => previousValue.data)
+			.reduce((previousValue, currentValue) => previousValue.concat(currentValue))
+			.filter(value => value.action === this.celestialRevolution)
 			.length
-	}
 
-	get chakras() {
-		return this.casts.filter(event => [this.data.actions.THE_FORBIDDEN_CHAKRA.id, this.data.actions.ENLIGHTENMENT.id].includes(event.action)).length
+		if (celestialRevolutionUsages > 0) {
+			return new Suggestion({
+				icon: this.celestialRevolution.icon,
+				content: <Trans id="mnk.rof.suggestions.celestialrevolution.content">Avoid using <ActionLink
+					action="CELESTIAL_REVOLUTION"/>.</Trans>,
+				severity: SEVERITY.MAJOR,
+				why: <Trans id="mnk.rof.suggestions.celestialrevolution.why"><ActionLink
+					action="RISING_PHOENIX"/> and <ActionLink action="ELIXIR_FIELD"/> do more potency along with
+					AoE.</Trans>,
+			})
+		}
 	}
 
-	get meditations() {
-		return this.casts.filter(event => event.action === this.data.actions.MEDITATION.id).length
+	output(windows: Array<HistoryEntry<EvaluatedAction[]>>): EvaluationOutput {
+		return {
+			format: 'table',
+			header: {
+				header: <ActionLink showName={false} action={'MASTERFUL_BLITZ'}/>,
```

Comment:
Lets be consistent, unless there's a reason not to.
```suggestion
				header: <ActionLink showName={false} action="MASTERFUL_BLITZ"/>,
```